ui/src/components/constants/constants.js

```diff
@@ -22,7 +22,7 @@ export const GROUP_MEMBER_NAME_REGEX =
 export const MICROSEGMENTATION_SERVICE_NAME_REGEX =
     '\\*|([a-zA-Z0-9_][a-zA-Z0-9_-]*\\.)*[a-zA-Z0-9_][a-zA-Z0-9_-]*';
 export const GROUP_ROLES_CATEGORY = 'group-roles';
-export const GROUP_MEMBER_PLACEHOLDER = 'user.<userid> or <domain>.<service>';
+export const GROUP_MEMBER_PLACEHOLDER = `${process.env.NEXT_PUBLIC_USER_DOMAIN}.<userid> or <domain>.<service>`;
```

Comment:
To avoid breaking the existing builds of UI, can you export a constant which would be set to `user` if `NEXT_PUBLIC_USER_DOMAIN` is undefined else would use the value of environment variable.
That constant can be used in other client files as well.

Reply:
Would it be better to define `USER_DOMAIN` in constants.js and reference it from the required components as follows?

constants.js

```js
export const USER_DOMAIN = process.env.NEXT_PUBLIC_USER_DOMAIN || 'user';
export const GROUP_MEMBER_PLACEHOLDER = `${USER_DOMAIN}. <userid> or <domain>. <service>`;
```

e.g.) Apply USER_DOMAIN to AddMember.js

```js
import { GROUP_MEMBER_NAME_REGEX, USER_DOMAIN } from '. /constants/constants';

... snip...

                            placeholder={
                                this.props.category === 'role' &&
                                this.props.collection ! == 'admin' !
                                    ? `${USER_DOMAIN}. <shortid> or <domain>. <service> or unix.<group> or <domain>:group.<group>`
                                    : `${USER_DOMAIN}. <shortid> or <domain>. <service> or unix.<group>`
```

Reply:
Yes, this looks good

Reply:
Something similar will also be required for server side code

Reply:
The server-side code uses the `userDomain` values defined in `src/config/config.js`.  Does this mean I should refer to `process.env.NEXT_PUBLIC_USER_DOMAIN` in config.js as well?

Reply:
I am sorry I didn't notice that userDomain is already defined. This shouldn't be an issue.